src/dune_pkg/opam_repo.ml

```diff
@@ -265,59 +274,110 @@ let load_opam_package t opam_package =
       { With_file.opam_file; file = Path.source @@ Path.Source.of_local file })
 ;;
 
+let load_packages_from_git rev_store opam_packages =
+  let+ contents = Rev_store.content_of_files rev_store opam_packages in
+  List.map2 opam_packages contents ~f:(fun file opam_file_contents ->
+    let path = Rev_store.File.path file in
+    let opam_file =
+      let filename =
+        (* the filename is used to read the version number *)
+        Path.Local.to_string path |> OpamFilename.of_string |> OpamFile.make
+      in
+      OpamFile.OPAM.read_from_string ~filename opam_file_contents
+    in
+    (* TODO the [file] here is made up *)
```

Comment:
I don't quite understand this TODO. What is the action here? Probably not to create the file, since we deliberately don't do that.

Reply:
The file is used as a location in case of an error. Unfortunately, this file is totally bogus since it only exists as a git object. I'm not sure how to write a proper location here yet unfortunately.

Reply:
If this location is only for error messages maybe we should potentially have our own error type here with its own `pp` function that could tell the user whether it is an path in their filesystem or a path in some revision of a repo? 

Anyway, this comment was not introduced in this PR nor has the fake file been introduced here so this is just a drive-by remark and can be improved at a later point in time.